test_app/tests/authentication/authenticator_plugins/test_azuread.py

```diff
@@ -23,6 +25,46 @@ def test_azuread_auth_successful(authenticate, unauthenticated_api_client, azure
     assert response.status_code == 200
 
 
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "key, secret, expected_status_code, expected_error",
+    [
+        (None, None, 400, {'KEY': ['This field may not be null.']}),
+        ('', None, 400, {'KEY': ['This field may not be blank.']}),
+        ('testaz', '', 400, {'SECRET': ['This field may not be blank.']}),
+        ('testaz', None, 201, {}),
+        ('testaz', "testaz_secret", 201, {}),
+    ],
+)
+def test_azuread_callback_url_validation(
+    admin_api_client,
+    key,
+    secret,
+    expected_status_code,
+    expected_error,
+):
+    config = {"KEY": key, "SECRET": secret}
+
+    data = {
+        "name": "AZUREAD TEST",
+        "enabled": True,
+        "create_objects": True,
+        "remove_users": True,
+        "configuration": config,
+        "type": "ansible_base.authentication.authenticator_plugins.azuread",
+    }
+
+    url = get_relative_url("authenticator-list")
+    response = admin_api_client.post(url, data=data, format="json")
+    assert response.status_code == expected_status_code
+    if expected_error:
+        assert response.json() == expected_error
+    else:
+        slug = response.data["slug"]
+        expected_path = get_fully_qualified_url('social:complete', kwargs={'backend': slug})
+        assert response.json()['configuration']['CALLBACK_URL'] == f"http://testserver{expected_path}"
```

Comment:
You can make get_fully_qualified_url auto-prepend a server by setting `FRONT_END_URL`. Something like this should work:

```suggestion
        slug = response.data["slug"]
        with override_settings(FRONT_END_URL='http://testserver/'):
            expected_path = get_fully_qualified_url('social:complete', kwargs={'backend': slug})
            assert response.json()['configuration']['CALLBACK_URL'] == expected_path
```

You would also need this import:
```
from django.test import override_settings
```

It would be bad to wrap your entire test in that as well because that should also effect the creation of the CALLBACK_URL so if the mock server changes from `http://testserver` to something else your function wouldn't care because of your override setting.